### ISSUE TITLE
CLI/Click design

### ISSUE THREAD

**Issue by StevenClontz** [StevenClontz/pretext.py#16]
As the CLI grows in complexity, we should make sure we're consistent in how it's configured for arguments/options. Here's what I propose:

- If the user is supplying required data that is not from a known finite list (e.g. book titles, filepaths, etc.), we use `@click.argument`
- If the user is supplying optional data that is not from a known finite list, we use `@click.option` with a sensible default as needed.
- If the user is supplying data from a known finite list (e.g. creating a new book|article), we use `@click.option` with a sensible default.

In that last case, we should decide between https://click.palletsprojects.com/en/7.x/options/#feature-switches and https://click.palletsprojects.com/en/7.x/options/#choice-options for non-boolean decisions.

What do you think @oscarlevin?

**Comment by daverosoff** [comment]
This seems like a good decision. So in the last case, the "feature switches" choice would be like this:
```bash
pretext --format=html
```
while the second looks like this:
```bash
pretext --html
```
Is that right? I would lean toward the more verbose, explicit option.

**Comment by StevenClontz** [comment]
Thanks @daverosoff -- and I think that provides better out-of-the-box feedback addressing #27 as well:

```
[pipenv]$ pretext build --help
Usage: pretext build [OPTIONS]

  Process PreTeXt files into specified format.

Options:
(...)
  --format [html|latex|all]  Sets which format to build  [default: html]
(...)

[pipenv]$ pretext build --format=foo
Usage: pretext build [OPTIONS]
Try 'pretext build --help' for help.

Error: Invalid value for '--format': invalid choice: foo. (choose from html, latex, all)
```

So my vote is for https://click.palletsprojects.com/en/7.x/options/#choice-options which is implemented in 9c76a45 -- if @oscarlevin approves we can close this and #27 I think.

**Comment by StevenClontz** [comment]
After discussion with Oscar, we're adopting this for 0.0:

- If the user is supplying data that is not from a known finite list (e.g. book titles, filepaths, etc.), we use `@click.option` with a sensible default.
- If the user is supplying data from a known finite list (e.g. creating a new book|article), we use `@click.option` with a sensible default and `click.Choice()`.
- Each command may have at most one `@click.argument` replacing a very important / most common option, with a sensible default.
